exe/CwTool.hs

```diff
@@ -1,68 +1,90 @@
 {-# LANGUAGE OverloadedStrings #-}
 module Main where
 
-import Configuration.Utils.CommandLine
-import Options.Applicative
-import Text.PrettyPrint.ANSI.Leijen (fillSep, text, vcat)
+import System.Environment
+import System.Exit
+import Text.Printf
 
 import qualified Chain2Gexf
 import qualified Ea
 import qualified RunNodes
 import qualified SlowTests
 import qualified TransactionGenerator
 
-data Command
-  = C2Gexf
-  | Ea
-  | GenTransactions
-  | RunNodes
-  | SlowTests
+main :: IO ()
+main = do
+    args <- getArgs
+    case args of
+      [] -> printHelp topLevelCommands
+      ["-h"] -> printHelp topLevelCommands
+      ["--help"] -> printHelp topLevelCommands
+      (cmd:restOfArgs) -> do
+        case filter (\cs -> cmd == csCmd cs) topLevelCommands of
+          [] -> do
+            printf "Error: \"%s\" is not a valid command\n\n" cmd
+            printHelp topLevelCommands
+            exitFailure
+          [cs] -> withArgs restOfArgs $ csAction cs
+          _ -> error "Duplicate command encountered.  This shouldn't happen!"
```

Comment:
I'm surprised that `optparse-applicative` can't "just do" what we were trying to do here.

Reply:
Yeah, I had the same thought.

https://github.com/pcapriotti/optparse-applicative/issues/340

I think we'd be able to do it if each subcommand was also an optparse-applicative parser that we had full control over (although I still think it would not be as composable as I was hoping).  But alas, that is not the case.  We don't have control over the `SlowTests` main because it just calls out to `defaultMain` from `tasty`.